transactions.go

```diff
@@ -0,0 +1,82 @@
+package amqp
+
+import (
+	"github.com/Azure/go-amqp/internal/buffer"
+	"github.com/Azure/go-amqp/internal/encoding"
+)
+
+// TransactionCapability represents capabilities for a transaction coordinator.
+type TransactionCapability encoding.Symbol
+
+const (
+	// LocalTransactionCapability indicates the transaction coordinator supports local Transactions.
+	LocalTransactionCapability TransactionCapability = "amqp:local-transactions"
+
+	// DistributedTransactionCapability indicates the transaction coordinator supports AMQP Distributed Transactions.
+	DistributedTransactionCapability TransactionCapability = "amqp:distributed-transactions"
+
+	// PromotableTransactionCapability indicates the transaction coordinator supports AMQP Promotable Transactions.
+	PromotableTransactionCapability TransactionCapability = "amqp:promotable-transactions"
+
+	// MultiTxnPerSessionTransactionCapability indicates the transaction coordinator supports multiple active transactions on a single session.
+	MultiTxnPerSessionTransactionCapability TransactionCapability = "amqp:multi-txns-per-ssn"
+)
+
+type TransactionDeclare struct {
+	GlobalID any
+}
+
+func (t TransactionDeclare) Marshal(wr *buffer.Buffer) error {
+	// TODO: add to typecode
+	return encoding.MarshalComposite(wr, encoding.TypeCodeTransactionDeclare, []encoding.MarshalField{
+		{Value: t.GlobalID, Omit: t.GlobalID == nil},
+	})
+}
+
+type TransactionDischarge struct {
+	TransactionID any
```

Comment:
Can it really be anything or is it constrained like `MessageID` is? If so, the constraints should be doc'ed.

Reply:
Interestingly, we do document the MessageID type, although it's also '*', so technically there's no constraint in the AMQP spec for it.

This ID is similar so I'm not sure it's up to us to document it, or how. I'm imagining it could be broker-specific.

Reply:
It looks like we can constrain this to be a []byte, which is great. However, we can't do a length check because our most common case (Service Bus) returns a length greater than 30.

Reply:
Changed it to a []byte, as per offline discussion.